cmd/create.go

```diff
@@ -132,7 +134,17 @@ func runCreate(cmd *cobra.Command, _ []string) error {
 		}
 	}
 
-	provisioner, err := getProvisioner(provider, accessToken, secretKey, organisationID, region)
+	var subscriptionID string
+	if provider == "azure" {
+		subscriptionID, _ = cmd.Flags().GetString("subscription-id")
+		authFile, _ := cmd.Flags().GetString("access-token-file")
+		err = os.Setenv("AZURE_AUTH_LOCATION", authFile)
```

Comment:
Can you rather pass the variable to the SDK than affect the OS environment variables?

Reply:
I had a look at the code, it does not seem to support passing as a variable: https://github.com/Azure/go-autorest/blob/master/autorest/azure/auth/auth.go#L295

Also their doc is using env for all the example, https://github.com/Azure/azure-sdk-for-go#authentication

So I'm not sure if passing as go variable instead of env is doable without changing the SDK. I will reach out to Azure GO SDK team.

Reply:
Can you do this differently then? Can you automatically set the value inside the provisioner, so that it's fully encapsulated for all users / calls.